addon/store.js

```diff
@@ -0,0 +1,48 @@
+import Ember from 'ember';
+import DS from 'ember-data';
+
+const keys = Ember.keys;
+const create = Ember.create;
+export const knownStores = [];
+
+export default DS.Store.extend({
+  init() {
+    this._super(...arguments);
+    knownStores.push(this._debugContainerKey);
```

Comment:
A given `store:<name>` should only be looked up once (it is a singleton in the container).  Each request gets a new container instance.  So each store (by name) should only be instantiated once.